config_example.txt
{
    "init_img_path": "",
    "input_images_path": "./input",
    "output_folder": "./output",
    "zoom_image_folder": "./zoom_images",
    "face_image_folder": "./face_images",
    "flow_image_folder": "./flow_images",

    "base_prompt": "(masterpiece,best quality:1.2),modelshoot style, (extremely detailed 8k wallpaper), block paint depicting a character in a cyberpunk street, posed character design study, backlit, light rays, highly detailed, trending on artstation",
    "face_prompt": "(masterpiece,best quality:1.2), face close up, ",
    "neg_prompt": " easynegative ng_deepnegative_v1_75t, text, watermark, nsfw, nude, ",

    "seed": -1,
    "seed_mode": "fixed",
    "sampler_name": "Euler a", 
    "sampler_step": 30,
    "face_sampler_step": 20,
    "cfg_scale": 7,
    "frame_width": 720,
    "frame_height": 1280,

    "interrogate_model": "clip",
    "use_base_img2img": true,
    "use_face_img2img": false,
    "use_zoom_img2img": false,

    "denoising_strength": 0.75,
    "face_denoising_strength": 0.5,
    "face_threshold": 0.2,
    "face_padding": 16,
    "zoom_denoising_strength": 0.5,

    "temporalnet": "v2",
    "temporalnet_weight": 0.4,
    "zoom_temporalnet": "v2",
    "zoom_temporalnet_weight": 0.4,
    "face_temporalnet": "v2",
    "face_temporalnet_weight": 0.4,

    "use_interrogate": false,
    "use_face_interrogate": false,
    
    "zoom_area_limit": 2211840, 
    "zoom_max_resolusion": 2048,
    "zoom_rects": [],
    
    "controlnet_lowvram": false,

    "controlnet": [
        {
            "module": "none",
            "model": "control_v11f1e_sd15_tile_fp16 [3b860298]",
            "weight": 1,
            "control_mode": 1,
            "pixel_perfect": true
        }, 
        {
            "module": "openpose",
            "model": "control_v11p_sd15_openpose_fp16 [73c2b67d]",
            "control_mode": 2,
            "pixel_perfect": true,
            "weight": 1
        }, 
        {
            "module": "softedge_hed",
            "model": "control_v11p_sd15_softedge_fp16 [f616a34f]",
            "weight": 0.4,
            "pixel_perfect": true,
            "control_mode": 2
        }
    ],
    "zoom_controlnet": [
        {
            "module": "none",
            "model": "control_v11f1e_sd15_tile_fp16 [3b860298]",
            "weight": 1,
            "control_mode": 1,
            "pixel_perfect": true
        }, 
        {
            "module": "openpose",
            "model": "control_v11p_sd15_openpose_fp16 [73c2b67d]",
            "control_mode": 2,
            "pixel_perfect": true,
            "weight": 1
        }, 
        {
            "module": "softedge_hed",
            "model": "control_v11p_sd15_softedge_fp16 [f616a34f]",
            "weight": 0.4,
            "pixel_perfect": true,
            "control_mode": 2
        }
    ],
    "face_controlnet": [
        {
            "module": "openpose_faceonly",
            "model": "control_v11p_sd15_openpose_fp16 [73c2b67d]",
            "control_mode": 2,
            "pixel_perfect": true,
            "weight": 1
        }
    ]
}



examples/config_mess.txt
{
    "checkpoint": "mistoonSapphire",
    "input_folder": "./input",
    "output_folder": "./output",
    "zoom_image_folder": "./zoom_images",
    "face_image_folder": "./face_images",
    "flow_image_folder": "./flow_images",

    "base_prompt": "(masterpiece,best quality, high quality:1.2), 8k uhd, three young women standing in front of a yellow background with a performance st sign behind them that says dance level <lora:3DMM_V12:1>",
    "face_prompt": "(masterpiece,best quality:1.2), face close up, ",
    "neg_prompt": "ng_deepnegative_v1_75t, (worst quality, low quality:1.2), text, watermark,  ",

    "seed": 4032950113,
    "seed_mode": "fixed",
    "sampler_name": "Euler a", 
    "sampler_step": 20,
    "face_sampler_step": 20,
    "cfg_scale": 2,
    "frame_width": 1080,
    "frame_height": 1920,
    "frame_crop": [0, 576, 1080, 1080],

    "interrogate_model": "clip",
    "use_base_img2img": true,
    "use_face_img2img": true,
    "use_zoom_img2img": false,

    "denoising_strength": 0.45,
    "face_denoising_strength": 0.4,
    "face_threshold": 0.2,
    "face_padding": 16,
    "zoom_denoising_strength": 0.65,

    "temporalnet": "v2",
    "temporalnet_weight": 0.3,
    "zoom_temporalnet": "v2",
    "zoom_temporalnet_weight": 0.3,
    "face_temporalnet": "v2",
    "face_temporalnet_weight": 0.3,

    "use_interrogate": false,
    "use_face_interrogate": false,
    
    "zoom_area_limit": 2211840, 
    "zoom_max_resolusion": 2048,
    "zoom_rects": [],
    
    "controlnet_lowvram": false,

    "controlnet": [
        {
            "module": "none",
            "model": "control_v11f1e_sd15_tile_fp16 [3b860298]",
            "weight": 0.8,
            "control_mode": 1,
            "pixel_perfect": true
        }, 
        {
            "module": "openpose",
            "model": "control_v11p_sd15_openpose_fp16 [73c2b67d]",
            "control_mode": 2,
            "pixel_perfect": true,
            "weight": 1
        }, 
        {
            "module": "softedge_hed",
            "model": "control_v11p_sd15_softedge_fp16 [f616a34f]",
            "weight": 0.4,
            "pixel_perfect": true,
            "control_mode": 2
        }
    ],
    "zoom_controlnet": [],
    "face_controlnet": [
        {
            "module": "none",
            "model": "control_v11f1e_sd15_tile_fp16 [3b860298]",
            "weight": 0.8,
            "control_mode": 1,
            "pixel_perfect": true
        }, 
        {
            "module": "openpose_faceonly",
            "model": "control_v11p_sd15_openpose_fp16 [73c2b67d]",
            "control_mode": 2,
            "pixel_perfect": true,
            "weight": 0.8
        }
    ]
}



helper/config.py
class Config(dict):
    def __init__(self, **kwargs):
        self.checkpoint = ""
        self.init_image_path = ""
        self.input_folder = "./input"
        self.output_folder = "./output"
        self.zoom_image_folder = "./zoom_images"
        self.face_image_folder = "./face_images"
        self.flow_image_folder = "./flow_images"

        self.base_prompt = "(masterpiece, high quality:1.2),"
        self.face_prompt = "(masterpiece, high quality:1.2), face close up,"
        self.neg_prompt = "(worst quality, low quality:1.2), nsfw, nude,"

        self.seed = -1
        self.seed_mode = "fixed"  # fixed,iter,random
        self.sampler_name = "Euler a"
        self.sampler_step = 30
        self.face_sampler_step = 20
        self.cfg_scale = 7
        self.frame_width = 1080
        self.frame_height = 1920
        self.frame_resize = "crop"  # crop,fit,resize
        self.frame_resize_anchor = "center"  # center, top
        self.frame_crop = None  # [x, y, width, height]
        self.frame_zoom = None  # [scale, offsetx, offsety, frames(0=immediately)]
        self.start_frame = 1
        self.dynamic_face_zoom = False

        self.interrogate_model = "clip"  # clip, deepdanbooru
        self.face_interrogate_model = "deepdanbooru"  # clip, deepdanbooru
        self.use_base_img2img = True
        self.use_face_img2img = False
        self.use_zoom_img2img = False

        self.denoising_strength = 0.5
        self.face_denoising_strength = 0.4
        self.face_threshold = 0.35
        self.face_padding = 16
        self.face_blur = 16
        self.zoom_blur = 16
        self.face_source = "input"
        self.zoom_denoising_strength = 0.65

        self.use_interrogate = False
        self.use_face_interrogate = False

        self.temporalnet_reset_interrogate = False
        self.temporalnet_reset_frames = []
        self.temporalnet = ""  # v1, v2
        self.temporalnet_weight = 0.3
        self.zoom_temporalnet = ""  # v1, v2
        self.zoom_temporalnet_weight = 0.3
        self.face_temporalnet = ""  # v1, v2
        self.face_temporalnet_weight = 0.3
        self.zoom_area_limit = 2048 * 1080
        self.zoom_max_resolusion = 2048
        self.zoom_rects = []  # [x, y, width, height, start_frame, end_frame]
        self.controlnet_lowvram = False
        self.controlnet = []
        self.zoom_controlnet = []
        self.face_controlnet = []
        self.frame_schedule = {}  # { "frame": config }

        for key, value in kwargs.items():
            setattr(self, key, value)


class PatchConfig(dict):
    def __init__(self, **kwargs):
        self.particle_folder = "./input"
        self.bg_folder = "./output"
        self.output_folder = "./patch"
        self.patch_rects = []  # [x, y, width, height, start_frame, end_frame]

        for key, value in kwargs.items():
            setattr(self, key, value)
